internal/pkg/cli/pipeline_init.go

```diff
@@ -326,42 +402,86 @@ func (o *initPipelineOpts) parseBitbucketRepoDetails() error {
 	o.repoName = repoDetails.name
 	o.repoOwner = repoDetails.owner
 
-	if o.repoBranch == "" {
-		o.repoBranch = defaultBBBranch
-	}
 	return nil
 }
 
 func (o *initPipelineOpts) selectURL() error {
-	// Fetches and parses all remote repositories.
-	err := o.runner.Run("git", []string{"remote", "-v"}, exec.Stdout(&o.buffer))
-	if err != nil {
-		return fmt.Errorf("get remote repository info: %w; make sure you have installed Git and are in a Git repository", err)
-	}
-	urls, err := o.parseGitRemoteResult(strings.TrimSpace(o.buffer.String()))
+	URLsAndRemoteNames, err := o.fetchAndParseURLs()
 	if err != nil {
 		return err
 	}
-	o.buffer.Reset()
+
+	var formattedRemoteNamesAndURLs []string
+	for _, repo := range URLsAndRemoteNames {
+		formattedRemoteNamesAndURLs = append(formattedRemoteNamesAndURLs, fmt.Sprintf(fmtRepoSelection, repo.remoteName, repo.remoteURL))
+	}
+	if len(formattedRemoteNamesAndURLs) == 1 {
```

Comment:
There are a few repetitive lines - the concern is that if we need to make changes to the implementation, we need to remember to change it at two places. What do you think of
```go
         var selectedRepo string
	if len(formattedRemoteNamesAndURLs) == 1 {
                log.Infoof()
                selectedRepo = formattedRemoteNamesAndURLs[0]
         } else {
                var err error
                selectedRepo, err = o.prompt.SelectOne()
                if err != nil {}
         }
         remote, url := o.parse(selectedRepo)
         // ... the rest
```

I feel like in this scenario reducing duplicate codes is preferred over returning early 🤔  but I could be wrong

Reply:
Yeah, good question. I feel like it may be a little easier to read as is, but I can see your point. I'll leave it as is for now, because we'll be making more changes to this file soon enough and we can reconsider then!